metaflow/plugins/pypi/pypi_decorator.py

```diff
@@ -64,7 +64,7 @@ def step_init(self, flow, graph, step, decos, environment, flow_datastore, logge
         # one here. We can consider introducing a UX where @pypi is able to consume
         # poetry.lock files in the future.
 
-        _supported_virtual_envs = ["conda"]  # , "venv"]
+        _supported_virtual_envs = ["conda", "docker"]  # , "venv"]
```

Comment:
@romain-intel this relates to discussions on decorator improvements. Should this be reliably extendable through extensions?

Main motivation is that this whole imagebakery feature could be a standalone extension, except for the few additions it needs to make to the OSS PyPI plugin.

Reply:
`_supported_virtual_envs` is just an implementation detail here, but what I'm pitching is adding a mechanism where a newly introduced environment could add itself to a whitelist for known decorators.